sdk/src/Core/Amazon.Runtime/ClientConfig.cs

```diff
@@ -221,7 +222,20 @@ public abstract string ServiceVersion
         public SigningAlgorithm SignatureMethod
         {
             get { return this.signatureMethod; }
-            set { this.signatureMethod = value; }
+            set { 
+                this.signatureMethod = value;
+                this.isSignatureMethodExplicitlySet = true;
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the SignatureMethod property was explicitly set by the user.
+        /// This is used for backwards compatibility to determine when legacy SignatureMethod configuration
+        /// should take precedence over auth scheme preferences.
+        /// </summary>
+        public bool IsSignatureMethodExplicitlySet
```

Comment:
I don't follow why this is needed... The naming is confusing but signature method is not going to be SigV4 / SigV4A, it's this enum instead: https://github.com/aws/aws-sdk-net/blob/0980c7ca63599f992adf1f7560a5caac8119906d/sdk/src/Core/Amazon.Runtime/Enumerations.cs#L24-L28